_posts/linda_xiang/2016-12-07-Cloud-data-transfer.md

```diff
@@ -0,0 +1,143 @@
+---
+layout: post
+title: "GitHub repository as job scheduling system to orchestrate large data transfer"
+breadcrumb: true
+author: junjun_zhang, linda_xiang
+date: 2016-12-12
+categories: linda_xiang
+tags:
+   - GitHub
+   - data transfer
+   - job scheduling
+teaser:
+   info: We use JSON files to represent data transfer jobs. Each JSON file contains necessary information for a data transfer job. JSON files are pre-generated and checked into a GitHub repository, which is structured with a few directories each representing a job state. Each data transfer worker clones the GitHub repository and picks up a queued job by moving the corresponding JSON file from 'queued-jobs' to 'downloading-jobs' directory. It then commits the change and pushes back to GitHub central repository. If more than one worker picked up the same job, a Git merge conflict will occur which elegantly avoids double scheduling. As the transfer job progresses to next steps, the worker will add log to the job JSON file and move it from one directory to the next, every file change and movement will be committed and pushed to GitHub.
+   image:  linda_xiang/data-transfer/github.png# optional
+header: 
+   version: small
+   title: Software Engineering Blog
+   image: header-logo-crop.png
+   icon: icon-blog
+---
+
+## The task
+
+The Pan-Cancer Analysis of Whole Genomes
+([PCAWG](https://dcc.icgc.org/pcawg)) project is a collaborative effort by
+the International Cancer Genome Consortium (ICGC) to identify common
+mutation pattens from over 2,800 cancer whole genomes with an emphasize in
+non-coding regions. Large amount of data were generated and stored in
+multiple GNOS repositories, core dataset includes:
```

Comment:
This one might be the right one although it's not under 'Annai' (http://annaisystems.com/):  https://annaisystems.zendesk.com/hc/en-us/sections/200575407-GNOS-and-GeneTorrent